nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,88 +144,126 @@ def run_conversion(
 
-            image_series_kwargs = dict(image_series_kwargs_list[j])
-            if check_regular_timestamps(ts=timestamps):
-                fps = get_movie_fps(movie_file=file)
-                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
-            else:
-                image_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
+        image_series_kwargs_list_updated, file_paths_list = _check_duplicates(image_series_kwargs_list)
+        if starting_times is not None:
+            assert isinstance(starting_times, list) and all(
+                [isinstance(x, float) for x in starting_times]
+            ), "Argument 'starting_times' must be a list of floats."
+            assert len(starting_times) == len(image_series_kwargs_list_updated), (
+                "starting times list length must be equal number of unique ImageSeries " "containers to write to nwb"
+            )
+        else:
+            warn("starting_times not provided, setting to 'None'")
+            starting_times = [None] * len(image_series_kwargs_list_updated)
 
+        for j, (image_series_kwargs, file_list) in enumerate(zip(image_series_kwargs_list_updated, file_paths_list)):
             if external_mode:
-                image_series_kwargs.update(format="external", external_file=[file])
+                with VideoCaptureContext(str(file_list[0])) as vc:
+                    fps = vc.get_movie_fps()
+                image_series_kwargs.update(
+                    starting_time=starting_times[j], rate=fps, format="external", external_file=file_list
+                )
             else:
+                file = file_list[0]
                 uncompressed_estimate = Path(file).stat().st_size * 70
                 available_memory = psutil.virtual_memory().available
-                if not chunk_data and uncompressed_estimate >= available_memory:
+                if not chunk_data and not stub_test and uncompressed_estimate >= available_memory:
                     warn(
                         f"Not enough memory (estimated {round(uncompressed_estimate/1e9, 2)} GB) to load movie file as "
                         f"array ({round(available_memory/1e9, 2)} GB available)! Forcing chunk_data to True."
                     )
                     chunk_data = True
 
-                total_frames = len(timestamps)
-                frame_shape = get_frame_shape(movie_file=file)
-                maxshape = [total_frames]
-                maxshape.extend(frame_shape)
+                with VideoCaptureContext(str(file)) as video_capture_ob:
+                    if stub_test:
+                        video_capture_ob.frame_count = 10
+                    total_frames = video_capture_ob.get_movie_frame_count()
+                    frame_shape = video_capture_ob.get_frame_shape()
+                    start_time = starting_times[j] if starting_times[j] is not None else 0.0
```

Comment:
```suggestion
```
With `NaN` as default, this is unneeded.

Reply:
it would make all timestamps as Nans also, I dont think that would be a good idea. 
If starting time is not specified, its better to still have timestamps but with an updated description of the `ImageSeries` like: https://github.com/catalystneuro/nwb-conversion-tools/blob/212fb41916d849f1a1594a3c7dcce0133b2641ea/nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py#L265